test/merger/test_general_merger.py

```diff
@@ -169,3 +171,200 @@ def test_no_merge_across_reset() -> None:
+    from opensquirrel.merger.general_merger import rearrange_barriers
+
+    rearrange_barriers(circuit.ir)
+    assert str(circuit) == expected_result
+
+
+@pytest.mark.parametrize(
+    ("circuit", "expected_result"),
+    [
+        (
+            (
+                CircuitBuilder(2)
+                .H(1)
+                .Rz(1, Float(2.332))
+                .H(0)
+                .Rx(0, Float(1.423))
+                .barrier(0)
+                .barrier(1)
+                .H(1)
+                .Rz(1, Float(2.332))
+                .H(0)
+                .Rx(1, Float(1.423))
+                .barrier(0)
+                .barrier(1)
+                .to_circuit()
+            ),
+            """version 3.0
+
+qubit[2] q
+
+Anonymous gate: BlochSphereRotation(Qubit[0], axis=[ 0.60376 -0.52053  0.60376], angle=-2.18173, phase=1.5708)
+Anonymous gate: BlochSphereRotation(Qubit[1], axis=[0.36644 0.85525 0.36644], angle=-1.72653, phase=1.5708)
+barrier q[0]
+barrier q[1]
+H q[0]
+Anonymous gate: BlochSphereRotation(Qubit[1], axis=[ 0.28903 -0.42028 -0.86013], angle=1.66208, phase=1.5708)
+barrier q[0]
+barrier q[1]
+""",
+        ),
+        (
+            CircuitBuilder(3)
+            .H(0)
+            .Ry(0, Float(2))
+            .barrier(2)
+            .barrier(0)
+            .barrier(1)
+            .H(0)
+            .H(1)
+            .Ry(1, Float(2))
+            .barrier(1)
+            .H(2)
+            .barrier(2)
+            .barrier(0)
+            .barrier(1)
+            .to_circuit(),
+            """version 3.0
+
+qubit[3] q
+
+Anonymous gate: BlochSphereRotation(Qubit[0], axis=[ 0.97706  0.      -0.21296], angle=3.14159, phase=1.5708)
+barrier q[2]
+barrier q[0]
+barrier q[1]
+H q[0]
+Anonymous gate: BlochSphereRotation(Qubit[1], axis=[ 0.97706  0.      -0.21296], angle=3.14159, phase=1.5708)
+H q[2]
+barrier q[1]
+barrier q[2]
+barrier q[0]
+barrier q[1]
+""",
+        ),
+        (
+            CircuitBuilder(2)
+            .H(0)
+            .Ry(0, Float(2))
+            .barrier(1)
+            .barrier(0)
+            .barrier(1)
+            .H(0)
+            .H(1)
+            .Ry(1, Float(2))
+            .barrier(1)
+            .to_circuit(),
+            """version 3.0
+
+qubit[2] q
+
+Anonymous gate: BlochSphereRotation(Qubit[0], axis=[ 0.97706  0.      -0.21296], angle=3.14159, phase=1.5708)
+barrier q[1]
+barrier q[0]
+barrier q[1]
+H q[0]
+Anonymous gate: BlochSphereRotation(Qubit[1], axis=[ 0.97706  0.      -0.21296], angle=3.14159, phase=1.5708)
+barrier q[1]
+""",
+        ),
+    ],
+    ids=["generic_case", "circuit_with_irregular_barrier_order", "repeating_barrier"],
+)
+def test_sticky_barriers(circuit: Circuit, expected_result: str) -> None:
+    circuit.merge_single_qubit_gates()
+    from opensquirrel.merger.general_merger import rearrange_barriers
+
+    rearrange_barriers(circuit.ir)
+    assert str(circuit) == expected_result
```

Comment:
```suggestion
def test_merge_single_qubit_gates_plus_rearrange_barriers(circuit: Circuit, expected_result: str) -> None:
    circuit.merge_single_qubit_gates()
    rearrange_barriers(circuit.ir)
    assert str(circuit) == expected_result
```

We shouldn't need to import `rearrange_barriers` because it's used in the other tests. In case we should, better to import it globally.